v2/slogger/rolling_file_appender/state.go

```diff
@@ -0,0 +1,64 @@
+package rolling_file_appender
+
+import (
+	"encoding/json"
+	"os"
+	"path/filepath"
+	"time"
+)
+
+type state struct {
+	LogStartTime time.Time
+}
+
+func newState(logStartTime time.Time) *state {
+	return &state{logStartTime}
+}
+
+func readState(path string) (*state, error) {
```

Comment:
I couldn't tell just by looking at the tests. Is there a test exercising this codepath. Specifically a test that calls `state.write` followed by `state.readState` and makes sure the `time.Time` was serialized/deserialized correctly?